pkg/kubeapiserver/authenticator/config.go

```diff
@@ -293,16 +301,29 @@ func newLegacyServiceAccountAuthenticator(keyfiles []string, lookup bool, apiAud
 // newServiceAccountAuthenticator returns an authenticator.Token or an error
 func newServiceAccountAuthenticator(iss string, keyfiles []string, apiAudiences authenticator.Audiences, serviceAccountGetter serviceaccount.ServiceAccountTokenGetter) (authenticator.Token, error) {
 	allPublicKeys := []interface{}{}
+	var socketFile string
 	for _, keyfile := range keyfiles {
-		publicKeys, err := certutil.PublicKeysFromFile(keyfile)
+		info, err := os.Stat(keyfile)
 		if err != nil {
 			return nil, err
 		}
-		allPublicKeys = append(allPublicKeys, publicKeys...)
+		switch mode := info.Mode(); {
+		case mode&os.ModeSocket != 0:
+			socketFile = keyfile
+			break
+		case mode.IsRegular():
+			publicKeys, err := certutil.PublicKeysFromFile(keyfile)
+			if err != nil {
+				return nil, err
+			}
+			allPublicKeys = append(allPublicKeys, publicKeys...)
+		}
 	}
 
-	tokenAuthenticator := serviceaccount.JWTTokenAuthenticator(iss, allPublicKeys, apiAudiences, serviceaccount.NewValidator(serviceAccountGetter))
-	return tokenAuthenticator, nil
+	if socketFile != "" {
```

Comment:
to avoid confusion, error out when both a socket and regular key files were provided

Reply:
Moved to a new flag